src/rosecommon/src/itemdb.cpp

```diff
@@ -0,0 +1,60 @@
+#include "itemdb.h"
+#include "connection.h"
+
+using namespace RoseCommon;
+
+ItemDatabase::ItemDatabase() {
+  for ( int index = 1; index < 15; ++index ) {
+    _database[index].reserve( 1000 ); // We shouldn't need more then 1000 entries off the bat.
+  }
+}
+
+ItemDatabase::~ItemDatabase() {
+  _database.clear();
```

Comment:
I don't think that's necessary

Reply:
I did this to ensure the memory is cleaned up.

Reply:
Isn't it cleaned in the map's destructor?

Reply:
I like to be explicit. don't assume that it will do it automatically. We compile with at least 3 different compilers, so to prevent any compiler differences I will be explicit.